services/wfm/src/main/java/org/openkilda/wfm/topology/isllatency/bolts/IslLatencyBolt.java

```diff
@@ -0,0 +1,153 @@
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.isllatency.bolts;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.messaging.info.InfoData;
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.messaging.info.event.IslOneWayLatency;
+import org.openkilda.messaging.info.event.IslRoundTripLatency;
+import org.openkilda.model.Isl;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.error.IllegalIslStateException;
+import org.openkilda.wfm.error.IslNotFoundException;
+import org.openkilda.wfm.error.SwitchNotFoundException;
+import org.openkilda.wfm.topology.AbstractTopology;
+import org.openkilda.wfm.topology.isllatency.LatencyAction;
+import org.openkilda.wfm.topology.isllatency.service.DecisionMakerService;
+import org.openkilda.wfm.topology.isllatency.service.IslLatencyService;
+import org.openkilda.wfm.topology.utils.KafkaRecordTranslator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Tuple;
+
+@Slf4j
+public class IslLatencyBolt extends AbstractBolt {
+    private final PersistenceManager persistenceManager;
+    private transient IslLatencyService islLatencyService;
+    private transient DecisionMakerService decisionMakerService;
+
+    public IslLatencyBolt(PersistenceManager persistenceManager) {
+        this.persistenceManager = persistenceManager;
+    }
+
+    @Override
+    protected void init() {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        islLatencyService = new IslLatencyService(transactionManager, repositoryFactory);
+        decisionMakerService = new DecisionMakerService(repositoryFactory);
+    }
+
+    @Override
+    protected void handleInput(Tuple input) throws Exception {
+        Message message = pullValue(input, KafkaRecordTranslator.FIELD_ID_PAYLOAD, Message.class);
+
+        if (message instanceof InfoMessage) {
+            InfoData data = ((InfoMessage) message).getData();
+            if (data instanceof IslRoundTripLatency) {
+                handleRoundTripIslLatency((IslRoundTripLatency) data);
+            } else if (data instanceof IslOneWayLatency) {
+                handleOneWayIslLatency((IslOneWayLatency) data);
+            } else {
+                unhandledInput(input);
+            }
+        } else {
+            unhandledInput(input);
+        }
+    }
+
+    private void handleRoundTripIslLatency(IslRoundTripLatency data) {
+        try {
+            Isl isl = islLatencyService.setIslLatencyBySourceEndpoint(
```

Comment:
I would prefer to update db latency only in the case when it's really change. Since we measure it in microseconds I'd introduce some e constant that will be a surrounding of current latency. And update db value if and only if mod(prev_latency - curent_latency) > e